### PR TITLE
Enhance the step that waits for the tunnel connection

### LINKED ISSUE #4690: `botanist.WaitUntilTunnelConnectionExists` step is failing right away on transient errors
**How to categorize this issue?**

/area quality
/kind bug
/priority 3

**What happened**:
We observe high amount of transient errors on Shoot creation that usually self-heal after several minutes. Few examples:

```
task "Waiting until the Kubernetes API server can connect to the Shoot workers" failed: could not forward to vpn-shoot pod: port forward connection could not be established within five seconds

-> Events:
* service-controller reported 3m48s ago (x5 over 5m3s): There are no available nodes for LoadBalancer
* service-controller reported 3m48s ago (x5 over 5m3s): Error syncing load balancer: failed to ensure load balancer: there are no available nodes for LoadBalancer service kube-system/vpn-shoot
```

```
task "Waiting until the Kubernetes API server can connect to the Shoot workers" failed: no running vpn-shoot pod found yet in the shoot cluster

-> Events:
* service-controller reported 3m10s ago: There are no available nodes for LoadBalancer
```

It seems that before https://github.com/gardener/gardener/pull/4183, we were retrying with timeout 15min until the tunnel connection is established.
https://github.com/gardener/gardener/blob/0c0a17e8b091bfac3c90ccd9d52055a7cb10e285/pkg/operation/botanist/waiter.go#L155-L166

After https://github.com/gardener/gardener/pull/4183, the func looks like 

https://github.com/gardener/gardener/blob/694a812212c962073ca732fb056953e1258b7cb0/pkg/operation/botanist/waiter.go#L53-L95

Currently if the first check for a tunnel connection fails, the func will check for vpn-shoot LoadBalancer events of type Warning, and if there are any, it will return err (fail the step).

**What you expected to happen**:
gardenlet to retry the check for a tunnel connection for 15min, and return an error only after the timeout is exceeded.

**How to reproduce it (as minimally and precisely as possible)**:
See above.

**Environment**:

- Gardener version: v1.31.3
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

### PR DESCRIPTION
**Pull request by ialidzhikov**
/area quality
/kind enhancement

**Which issue(s) this PR fixes**:
Fixes #4690

**Release note**:

```other operator
`botanist.WaitUntilTunnelConnectionExists` step is now enhanced to do not fail right away when the tunnel connection cannot be established. Instead, the step will retry tunnel connection check until the configured timeout is exceeded.
```

### CODE REVIEW

File: pkg/operation/botanist/waiter.go [hunk 1]
```diff
@@ -82,16 +77,20 @@ func (b *Botanist) WaitUntilTunnelConnectionExists(ctx context.Context) error {
 			eventsErrorMessage, err2 := kutil.FetchEventMessages(ctx, b.K8sShootClient.Client().Scheme(), b.K8sShootClient.Client(), service, corev1.EventTypeWarning, 2)
 			if err2 != nil {
 				b.Logger.Errorf("error %v occurred while fetching events for VPN load balancer service", err2)
-				return retry.SevereError(fmt.Errorf("'%w' occurred but could not fetch events for more information", err))
+				return fmt.Errorf("'%w' occurred but could not fetch events for more information", err)
 			}
 
 			if eventsErrorMessage != "" {
-				return retry.SevereError(fmt.Errorf("%s\n\n%s", err.Error(), eventsErrorMessage))
```
Comment (rfranzke) [inline pkg/operation/botanist/waiter.go:89]: Wasn't it enough to just change this to `retry.MinorError`?

Comment (ialidzhikov) [reply pkg/operation/botanist/waiter.go:89]: If we change to `retry.MinorError`, then we will fetch the Service events on each retry until the timeout is exceeded. Why would we want to do this when we can fetch the events only once after the timeout is exceeded?

Comment (rfranzke) [reply pkg/operation/botanist/waiter.go:89]: I see, thanks for explaining.

**Review by vpnachev — APPROVED** [review]
/lgtm

**Pull request merged**